File: packages/shared/types/report.ts
import type { RiskLevel } from './analysis'

export interface ReportMetadata {
  report_id: string
  analysis_id: string
  file_name: string
  sha256: string
  generated_at: string
  risk_level: RiskLevel
  analyst: string
}

export type ExportFormat = 'pdf' | 'json' | 'ioc_txt'

export interface ReportExport {
  report_id: string
  format: ExportFormat
  download_url: string
}

export interface ReportGenerateRequest {
  analysis_id: string
  analyst: string
  formats: ExportFormat[]
}

export interface ReportGenerateResponse {
  report_id: string
  metadata: ReportMetadata
  download_urls: Record<ExportFormat, string>
}

File: backend/models/report.py
from pydantic import BaseModel, ConfigDict


class ReportMetadata(BaseModel):
    model_config = ConfigDict(extra='ignore')

    report_id: str
    analysis_id: str
    file_name: str
    sha256: str
    generated_at: str
    risk_level: str
    analyst: str


class ReportGenerateRequest(BaseModel):
    model_config = ConfigDict(extra='ignore')

    analysis_id: str
    analyst: str
    formats: list[str]


class ReportExport(BaseModel):
    model_config = ConfigDict(extra='ignore')

    report_id: str
    format: str
    download_url: str
